src/java.base/share/classes/jdk/internal/module/ModuleBootstrap.java

```diff
@@ -869,6 +874,23 @@ private static void addIllegalAccess(String illegalAccess,
         builder.complete();
     }
 
+    /**
+     * Process the --enable-native-access option to grant access to restricted methods to selected modules.
+     */
+    private static void addEnableNativeAccess(ModuleLayer layer) {
+        // add native modules explicitly provided on the command line
+
+        for (String name : IllegalNativeAccessChecker.enableNativeAccessModules()) {
+            Optional<Module> module = layer.findModule(name);
+            if (module.isPresent()) {
+                SharedSecrets.getJavaLangAccess().addEnableNativeAccess(module.get());
+            } else {
+                // silently skip.
+                // warnUnknownModule(ENABLE_NATIVE_ACCESS, name);
```

Comment:
Maybe for later but the other options do emit a warning when unknown module is specified. If the decoding of this command line option is moved to ModuleBootstrap then most of this class will go away and you will be able to use warnUnknownModule.